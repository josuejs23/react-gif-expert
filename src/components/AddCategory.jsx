import PropTypes from 'prop-types'
import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChanged = ({ target }) => {
        console.log(target.value)
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        if(inputValue.trim()===''){
            return;
        }
        onNewCategory(inputValue.trim())
        setInputValue('');
        console.log('FormValue',inputValue);
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type='text'
                placeholder='search category..'
                value={inputValue}
                onChange={onInputChanged}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory:PropTypes.func.isRequired
}