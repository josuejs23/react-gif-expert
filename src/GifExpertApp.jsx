import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ()=>{
    
    const [categories, setCategories] = useState([])

    const onAddCategory = (newCategory)=>{
        if(categories.includes(newCategory)) return;
        console.log(newCategory)
        setCategories([newCategory, ...categories]);
    }

    console.log(categories)
    return(
        <>
            <h1>Gif-Expert-App</h1>

            <AddCategory 
                onNewCategory={ (category)=>onAddCategory(category)}
            />
            
            {
                categories.map( (categorie )=> (
                    <GifGrid category={categorie} key={categorie}/>
                )
                )
            }
        </>
    )
}