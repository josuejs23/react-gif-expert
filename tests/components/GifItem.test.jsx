import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Test on <GifItem /> component', ()=>{
    
    const title = 'Any title';
    const url = 'https://imagenes.com/image.gif';

    test('Should match with the snapshot', ()=>{
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('Should have the url the image and the ALT', ()=>{
        render(<GifItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img'); 
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Should show the title at the component', ()=>{
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })

})