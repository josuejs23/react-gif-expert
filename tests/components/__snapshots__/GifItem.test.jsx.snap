// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifItem /> component Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Any title"
      src="https://imagenes.com/image.gif"
    />
    <p>
      Any title
    </p>
  </div>
</div>
`;
